src/Controller/AnnouncementController.php

```diff
@@ -569,25 +618,23 @@ public function editAction(
             if (!$announcementItem) {
                 throw $this->createNotFoundException('No announcement found for id ' . $roomId);
             }
-            $itemController = $this->get('commsy.item_controller');
```

Comment:
Controllers shouldn't call or inject themselves, the methods called here should be moved to itemService